### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by octopusinvitro**
This PR does a couple of small fixes:
- Fixes a bug where emails were sent in tests
- Removes a couple of nil checks
- Excludes a file from a Rubocop rule that was added in a last Rubocop version
- Updates readme
- Updates changelog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ If there are no errors, it means that:
 The `data` that the gem consumes can be anything that responds to at least the following (all these should return strings except the giftaid, which is a boolean):
 
 * `data.amount`: The amount of money to be donated, in currency units. This can be an amount with cents, and will be converted to cents by the gem. For example: 10.50
-* `data.currency`: The currency that the amount is in, should be a valid ISO 4217 code for currencies and supported by the Money gem that this gem uses to handle money. [Check here for a list of supported currencies in JSON format]((https://github.com/RubyMoney/money/tree/e2773f7859b268965fa003e2630ed58e7e96ac58/config)
+* `data.currency`: The currency that the amount is in, should be a valid ISO 4217 code for currencies and supported by the Money gem that this gem uses to handle money. [Check here for a list of supported currencies in JSON format](https://github.com/RubyMoney/money/tree/e2773f7859b268965fa003e2630ed58e7e96ac58/config)
```
Comment (ketan-survival) [inline README.md]: Reworded:
The currency that the amount is in, should be a valid ISO 4217 code for currencies and supported by the Money gem, which is used by this gem to handle money.

File: README.md [hunk 2]
```diff
@@ -131,12 +131,13 @@ You need to set your environment by running the credentials. These will be run w
 . credentials/.email_server
 ```
 
-At the moment, when you run the tests, any HTTP request will be recorded once in `spec/cassettes`, and then will be replayed everytime you run the tests. So the first time that you run them, they will take long, but once they are recorded, they should run faster.
+At the moment, when you run the tests, any HTTP request will be recorded once in `spec/cassettes`, and then will be replayed everytime you run the tests. So the first time that you run them, they will take long, but once they are recorded, they should run faster. Also, the first time you run the tests, you need to be connected to the internet.
```
Comment (ketan-survival) [inline README.md]: Reworded:

Currently, when you run the tests, HTTP request will be recorded once in `spec/cassettes`, which are then replayed every time you run the tests. So the first time that you run them, they will take a while, but once they are recorded they will run faster. Also, the first time you run the tests, you need to be connected to the internet.

File: README.md [hunk 3]
```diff
@@ -131,12 +131,13 @@ You need to set your environment by running the credentials. These will be run w
...
 
-If you ever get any weird error when running the tests, try removing the cassettes:
+If you ever get any weird error when running the tests, try removing the cassettes and then run the tests again:
```
Comment (ketan-survival) [inline README.md]: Reworded:

If you get strange errors when running the tests, try removing the cassettes and then running the tests again:

**Comment by octopusinvitro** [comment]
Merging, as feedback was addressed

**Pull request merged**